solidity/test/relay/LightRelayMaintainerProxy.test.ts

```diff
@@ -0,0 +1,532 @@
+
+  return {
+    deployer,
+    governance,
+    maintainer,
+    thirdParty,
+    reimbursementPool,
+    lightRelayMaintainerProxy,
+    lightRelay,
+  }
+}
+
+describe("LightRelayMaintainerProxy", () => {
+  let deployer: SignerWithAddress
+  let governance: SignerWithAddress
+  let maintainer: SignerWithAddress
+  let thirdParty: SignerWithAddress
+  let reimbursementPool: ReimbursementPool
+  let lightRelayMaintainerProxy: LightRelayMaintainerProxy
+  let lightRelay: LightRelay
+
+  before(async () => {
+    // eslint-disable-next-line @typescript-eslint/no-extra-semi
+    ;({
+      deployer,
+      governance,
+      maintainer,
+      thirdParty,
+      reimbursementPool,
+      lightRelayMaintainerProxy,
+      lightRelay,
+    } = await waffle.loadFixture(fixture))
+
+    await deployer.sendTransaction({
+      to: reimbursementPool.address,
+      value: ethers.utils.parseEther("100"),
+    })
+  })
+
+  describe("authorize", () => {
+    context("When called by non-owner", () => {
+      it("should revert", async () => {
+        await expect(
+          lightRelayMaintainerProxy
+            .connect(thirdParty)
+            .authorize(maintainer.address)
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when called by the owner", () => {
+      context("When the maintainer is already authorized", () => {
+        before(async () => {
+          await createSnapshot()
+
+          // Authorize the maintainer to see if the next attempt reverts.
+          await lightRelayMaintainerProxy
+            .connect(governance)
+            .authorize(maintainer.address)
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should revert", async () => {
+          await expect(
+            lightRelayMaintainerProxy
+              .connect(governance)
+              .authorize(maintainer.address)
+          ).to.be.revertedWith("Maintainer is already authorized")
+        })
+      })
+
+      context("When the maintainer is not authorized yet", () => {
+        let tx: ContractTransaction
+
+        before(async () => {
+          await createSnapshot()
+
+          tx = await lightRelayMaintainerProxy
+            .connect(governance)
+            .authorize(maintainer.address)
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should authorize the address", async () => {
+          expect(
+            await lightRelayMaintainerProxy.isAuthorized(maintainer.address)
+          ).to.be.true
+        })
+
+        it("should emit the MaintainerAuthorized event", async () => {
+          await expect(tx)
+            .to.emit(lightRelayMaintainerProxy, "MaintainerAuthorized")
+            .withArgs(maintainer.address)
+        })
+      })
+    })
+  })
+
+  describe("deauthorize", () => {
+    context("When called by non-owner", () => {
```

Comment:
```suggestion
    context("when called by non-owner", () => {
```

Reply:
Done in 462925b1bd07e05f87beb7c9fcba541d335c0be1.